### ISSUE TITLE
Deprecate in favour of Lombok

### ISSUE THREAD

**Issue by dwijnand** [playframework/play-enhancer#18]
... if possible.  Might need to check in the face of Twirl's code generation and compilation.

**Comment by mkurz** [comment]
That won't work because it's not possible to access lombok generated code from within Scala code.
More information can be found in my old comment here: https://github.com/playframework/playframework/pull/4006#issuecomment-81049150

**Comment by mkurz** [comment]
The Scala compiler doesn't know anything about what lombok annotations will generated, hence it can't access these generated methods:
https://github.com/scala/scala/tree/2.13.x/src/compiler/scala/tools/nsc/javac

**Comment by mkurz** [comment]
[Here](https://stackoverflow.com/questions/11171631/error-compiling-java-scala-mixed-project-and-lombok/11172044#11172044) is also a good explanation. The problem is the Scala parser does not support/run an annotation processor.

**Comment by dwijnand** [comment]
As feared.  Thanks for the wonderful explanation and links, @mkurz!

**Comment by marcospereira** [comment]
Thanks, @mkurz!

Well, play-enhancer stills not a good solution, and I think we need to deprecate it anyway even if Lombok is not a drop-in replacement. Maybe this issue is about documenting the alternatives (write the code yourself, use Lombok with a subproject).

**Comment by mkurz** [comment]
Yeah. I personally use Lombok within subprojects for my Play-Java projects. That works fine.

**Comment by marcospereira** [comment]
I've reopened and edited the title and description.

**Comment by mkurz** [comment]
Or just wait until Java itself catches up :laughing: :rofl:  (like https://cr.openjdk.java.net/~briangoetz/amber/datum.html)

**Comment by ignasi35** [comment]
> Or just wait until Java itself catches up

_buys a comfortable chair_

**Comment by marcospereira** [comment]
One thing that we can do just right now is to avoid using it in our sample applications. Not a 100% sure, but I think https://github.com/playframework/play-java-ebean-example is the only one using it.

**Comment by ignasi35** [comment]
A quick, _naïve_ search in my local repos:

```
find . | grep plugins.sbt | xargs grep enhanc| awk -F\/  '{print $2 }' | uniq
play-java-fileupload-example
play-java-grpc-example
play-java-jpa-example
play-java-starter-example
playframework
```

I wouldn't be suprised if it's used somewhere else.

PS: I don't see it used in [playframework/play-java-ebean-example](https://github.com/playframework/play-java-ebean-example/blob/2.6.x/project/plugins.sbt)

**Comment by dwijnand** [comment]
From https://github.com/search?l=Scala&q=org%3Aplayframework+sbt-play-enhancer+-repo%3Aplayframework%2Fplay-enhancer&type=Code:

* playframework/play-ebean
* playframework/play-generated-docs
* playframework/play-java-fileupload-example
* playframework/play-java-grpc-example
* playframework/play-java-jpa-example
* playframework/play-java-starter-example
* playframework/play-quota-java-example
* playframework/playframework

**Comment by marcospereira** [comment]
Yeah, for most of these samples, I don't think it is necessary. But I think play-ebean requires it, so we can just remove from all the other examples not using Ebean and use alternatives.
